src/Symfony/Component/HttpKernel/DataCollector/EventDataCollector.php

```diff
@@ -28,94 +28,60 @@
  */
 class EventDataCollector extends DataCollector implements LateDataCollectorInterface
 {
-    private ?EventDispatcherInterface $dispatcher;
+    /** @var iterable<EventDispatcherInterface> */
+    private iterable $dispatchers;
     private ?RequestStack $requestStack;
     private ?Request $currentRequest = null;
 
-    public function __construct(EventDispatcherInterface $dispatcher = null, RequestStack $requestStack = null)
+    /**
+     * @param iterable<EventDispatcherInterface>|EventDispatcherInterface|null $dispatchers
+     */
+    public function __construct(iterable|EventDispatcherInterface|null $dispatchers = [], RequestStack $requestStack = null)
     {
-        $this->dispatcher = $dispatcher;
+        if (!is_iterable($dispatchers)) {
+            trigger_deprecation('symfony/http-kernel', '6.3', 'Passing "null" or a single "%s" to "%s()" is deprecated, pass an "%1$s" iterable instead.', EventDispatcherInterface::class, __METHOD__);
```

Comment:
I feel like we don't need this deprecate since it's pretty easy to support all kinds of arguments

Reply:
Don’t really have an opinion on this one; should I remove the deprecation?